walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,108 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
 
+/** Class that lists possible meeting times based on meeting information it takes in. */
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+  private static final int END_OF_DAY = TimeRange.getTimeInMinutes(23, 59);
+
+  private static final Comparator<Event> ORDER_BY_START =
+      new Comparator<Event>() {
+        @Override
+        public int compare(Event a, Event b) {
+          return Long.compare(a.getWhen().start(), b.getWhen().start());
+        }
+      };
+
+  /**
+   * Returns a list of time periods in which the meeting, specified by request, could happen.
+   *
+   * @param eventsCollection the events we know about
+   * @param request information about the meeting, including attendees, optional attendees, and how
+   *     long it needs to be
+   */
+  public Collection<TimeRange> query(Collection<Event> eventsCollection, MeetingRequest request) {
+    HashSet<String> requestAttendees = new HashSet<String>(request.getAttendees());
+    ArrayList<Event> events =
+        getRelevantEvents(requestAttendees, new ArrayList<Event>(eventsCollection));
+    List<TimeRange> possibleMeetingTimes = new ArrayList<TimeRange>();
+    if (events.isEmpty()) {
```

Comment:
Nit: Can you add a comment describing the semantics of "events.isEmpty()" (e.g. what is the implication for the business logic)